pennylane/resource/first_quantization.py

```diff
@@ -0,0 +1,600 @@
+            error (float): target error in the algorithm
+            br (int): number of bits for ancilla qubit rotation
+            charge (int): total electric charge of the system
+
+        Returns:
+            int: the number of Toffoli gates needed to implement the first quantization algorithm
+
+        **Example**
+
+        >>> n = 100000
+        >>> eta = 156
+        >>> omega = 169.69608
+        >>> error = 0.01
+        >>> gate_cost(n, eta, omega, error)
+        3676557345574
+        """
+        if n <= 0:
+            raise ValueError("The number of plane waves must be a positive number.")
+
+        if eta <= 0 or not isinstance(eta, int):
+            raise ValueError("The number of electrons must be a positive integer.")
+
+        if omega <= 0:
+            raise ValueError("The unit cell volume must be a positive number.")
+
+        if error <= 0.0:
+            raise ValueError("The target error must be greater than zero.")
+
+        if br <= 0 or not isinstance(br, int):
+            raise ValueError("br must be a positive integer.")
+
+        if not isinstance(charge, int):
+            raise ValueError("system charge must be an integer.")
+
+        e_cost = FirstQuantization.estimation_cost(n, eta, omega, error, br=br, charge=charge)
+        u_cost = FirstQuantization.unitary_cost(n, eta, omega, error, br, charge)
+
+        return e_cost * u_cost
+
+    @staticmethod
+    def qubit_cost(n, eta, omega, error, br=7, charge=0):
+        r"""Return the number of logical qubits needed to implement the first quantization
+        algorithm.
+
+        The expression for computing the cost is taken from Eq. (101) of
+        [`arXiv:2204.11890v1 <https://arxiv.org/abs/2204.11890v1>`_].
+
+        Args:
+            n (int): number of plane waves
+            eta (int): number of electrons
+            omega (float): unit cell volume
+            error (float): target error in the algorithm
+            br (int): number of bits for ancilla qubit rotation
+            charge (int): total electric charge of the system
+
+        Returns:
+            int: number of logical qubits needed to implement the first quantization algorithm
+
+        **Example**
+
+        >>> n = 100000
+        >>> eta = 156
+        >>> omega = 169.69608
+        >>> error = 0.01
+        >>> qubit_cost(n, eta, omega, error)
+        4377
+        """
+        if n <= 0:
+            raise ValueError("The number of plane waves must be a positive number.")
+
+        if eta <= 0 or not isinstance(eta, int):
+            raise ValueError("The number of electrons must be a positive integer.")
+
+        if omega <= 0:
+            raise ValueError("The unit cell volume must be a positive number.")
+
+        if error <= 0.0:
+            raise ValueError("The target error must be greater than zero.")
+
+        if not isinstance(charge, int):
+            raise ValueError("system charge must be an integer.")
```

Comment:
I see a Codecov box here, is this case covered by the tests?

Reply:
This [line](https://github.com/PennyLaneAI/pennylane/pull/2644/files/6294a6d812b8ef6481a8229940525ea5020c7907#diff-d830cf664dc2d8efcb6875b068e210cd016651ba50ea2e4cd37546bd425b489aR198) should cover that condition.